compiler/noirc_frontend/src/hir/resolution/errors.rs

```diff
@@ -94,6 +94,8 @@ pub enum ResolverError {
     NoPredicatesAttributeOnUnconstrained { ident: Ident },
     #[error("#[fold] attribute is only allowed on constrained functions")]
     FoldAttributeOnUnconstrained { ident: Ident },
+    #[error("Interpreting array-length failed")]
+    ArrayLengthInterpreter { error: InterpreterError, span: Span },
```

Comment:
Can you rename this error to something which doesn't reveal/specify to users that we're interpreting array lengths? E.g. `InvalidArrayLength` or similar?